### PR TITLE
Refactor Base/Text component to TypeScript

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR updates the Base Text component to typescript

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Base/Text.tsx [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+  small?: boolean;
+  big?: boolean;
+  bigger?: boolean;
+  upper?: boolean;
+  link?: boolean;
+  strikethrough?: boolean;
+  underline?: boolean;
+}
+
+const createStyles = (colors: Colors) =>
```
Comment (gantunesr) [inline app/components/Base/Text.tsx:33]: Can you move the styles to a `styles.ts` file?

**Review by gantunesr — COMMENTED** [review]
Looks good, just one comment since you're refactoring this component

File: app/components/Base/Text.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/Base/Text.tsx]: Done in [`58e84cf` (#5429)](https://github.com/MetaMask/metamask-mobile/pull/5429/commits/58e84cf7de984f89ae3f40e7ad0696fc6167f644)

**Review by owencraston — APPROVED** [review]
Nice. In the future I think you van use the [text in from the component-library](https://github.com/MetaMask/metamask-mobile/blob/main/app/component-library/components/Texts/Text/README.md)

**Pull request merged**